### ISSUE TITLE
No support for environment variables from yarnrc.yml

### ISSUE THREAD

**Issue by kamronbatman** [raineorshine/npm-check-updates#1216]
- [X] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [X] I am using the latest version of `npm-check-updates`
- [X] I am using `node >= 14.14`

---

## Steps to Reproduce

Add a custom npmRegistryServer, like this:

```yml
npmScopes:
  mycompanynamespace:
    npmRegistryServer: "${MY_COMPANY_NPM_REGISTRY-https://artifactory.internal/artifactory/npm/npm-all}"
```

Steps:

Run `ncu` with, and without the environment variable set.

## Current Behavior

ncu throws an error because it can't determine the server.

## Expected Behavior

ncu should parse properly and use that registry server.

**Comment by raineorshine** [comment]
Hi, thanks for the suggestion. This looks straightforward enough. 

I'm open to PR's, or I can have a try when I have some free time in the future. 

Reference: https://yarnpkg.com/configuration/yarnrc#npmRegistryServer

**Comment by raineorshine** [comment]
Fixed and published in `v16.3.23`.

There was existing logic for interpolating yarn config values with env variables, it just wasn't being applied to `npmRegistryServer`.

**Comment by kamronbatman** [comment]
Thank you so much, and I am sorry I didn't have time to contribute! I have been using this for years and I love it!
